sr_port/mupip_backup.c

```diff
@@ -610,8 +610,12 @@ void mupip_backup(void)
 				mubclnup(rptr, need_to_del_tempfile);
 				mupip_exit(ustatus);
 			}
-			/* Creating temp filename here. Check if we have the required space. */
-			nbytes = SNPRINTF(tempfilename, SIZEOF(tempfilename), "%.*s/%.*s_%4x_XXXXXX",
+			/* Creating temp filename here. Check if we have the required space.
+			 * Use %04x (instead of %x) for process_id to ensure we have 4 bytes always even if process_id
+			 * is a small number (< 2**24 etc.) and is 0-filled on the left if needed. Makes the length
+			 * deterministic instead of being dependent on the process_id.
+			 */
+			nbytes = SNPRINTF(tempfilename, SIZEOF(tempfilename), "%.*s/%.*s_%04x_XXXXXX",
```

Comment:
Does it matter that we might be truncating a pid here? PIDs are *usually* < 32K but the pid_t itself is an int and I thought there was a configuration option on the Linux kernel to widen the used PID field once storage passed a certain point or some such thing - can't remember. Anyway, should this be 8x instead of 4x to properly represent a 32 bit value?

Reply:
Good catch. I forgot that hex digits are twice the byte length so it should be 8x instead of 4x. Will take care of that.